src/dipdup/database.py

```diff
@@ -125,157 +142,158 @@ def iter_models(package: Optional[str]) -> Iterator[tuple[str, Type[TortoiseMode
+    sql_path = Path(__file__).parent / 'sql' / 'truncate_schema.sql'
+    await execute_sql(conn, sql_path, schema_name, immune_schema_name)
+
+    # NOTE: Move immune tables to a separate schema - it's free!
+    if immune_tables:
+        await pg_create_schema(conn, immune_schema_name)
+        for table in immune_tables:
+            await pg_move_table(conn, table, schema_name, immune_schema_name)
+
+    await conn.execute_script(f"SELECT truncate_schema('{schema_name}')")
+
+    if immune_tables:
+        for table in immune_tables:
+            await pg_move_table(conn, table, immune_schema_name, schema_name)
+        await pg_drop_schema(conn, immune_schema_name)
 
-    await conn.execute_script(_truncate_schema_path.read_text())
-    await conn.execute_script(f"SELECT truncate_schema('{name}')")
+
+async def _wipe_schema_sqlite(
+    conn: SqliteClient,
+    path: str,
+    immune_tables: set[str],
+) -> None:
+    if path == ':memory:':
+        raise FrameworkException('Attempted to wipe in-memory database; that makes no sense')
+
+    # NOTE: Dropping huge tables and deleting from them is slow and I/O heavy in SQLite. It's better to save the tables of interest
+    # elsewhere and drop the whole database. First, create a new database and attach it to the current connection:
+    immune_path, namespace = f'{path}.immune', 'immune'
+    await conn.execute_script(f'ATTACH DATABASE "{immune_path}" AS {namespace}')
+
+    # NOTE: Copy immune tables to the new database.
+    master_query = 'SELECT name, type FROM sqlite_master'
+    result = await conn.execute_query(master_query)
+    for name, type_ in result[1]:
+        if type_ != 'table' or name not in immune_tables:
+            continue
+
+        expr = f'CREATE TABLE {namespace}.{name} AS SELECT * FROM {name}'
+        _logger.info('Executing `%s`', expr)
+        await conn.execute_script(expr)
+
+    # NOTE: Now the weirdest part - swap the databases when the program exits and all connections are closed.
+    def _finish_wipe() -> None:
+        _logger.info('Restoring immune tables')
+        Path(immune_path).replace(path)
+
+    atexit.register(_finish_wipe)
 
 
 async def wipe_schema(
-    conn: BaseDBAsyncClient,
+    conn: SupportedClient,
     schema_name: str,
-    immune_tables: Set[str],
+    immune_tables: set[str],
 ) -> None:
     """Truncate schema preserving immune tables. Executes in a transaction"""
-    if isinstance(conn, SqliteClient):
-        raise NotImplementedError
-
-    immune_schema_name = f'{schema_name}_immune'
-
     async with conn._in_transaction() as conn:
-        if immune_tables:
-            await create_schema(conn, immune_schema_name)
-            for table in immune_tables:
-                await move_table(conn, table, schema_name, immune_schema_name)
-
-        await truncate_schema(conn, schema_name)
+        if isinstance(conn, SqliteClient):
+            await _wipe_schema_sqlite(conn, schema_name, immune_tables)
+        else:
```

Comment:
<nerd mode>
```
else if isinstance(conn, AsyncpgClient):
   ...
else:
   raise NotImplementedError(conn)
```

Reply:
Nah, that's exact kinda code review I pursue.